catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/config/ConfigurationApplication.java

```diff
@@ -212,6 +213,68 @@
 
   private String customBackgroundSlideout;
 
+  private List<String> editorAttributes = Collections.emptyList();
+  private List<String> requiredAttributes = Collections.emptyList();
+  private Map<String, Set<String>> attributeEnumMap = Collections.emptyMap();
+
+  public List<String> getEditorAttributes() {
+    return editorAttributes;
+  }
+
+  public void setEditorAttributes(List<String> editorAttributes) {
+    this.editorAttributes = editorAttributes;
+  }
+
+  public List<String> getRequiredAttributes() {
+    return requiredAttributes;
+  }
+
+  public void setRequiredAttributes(List<String> requiredAttributes) {
+    this.requiredAttributes = requiredAttributes;
```

Comment:
❓ Should the list of required attributes be validated against the list of editable attributes?

Reply:
The way it works right now, required attributes are ignored if they're not shown in the editor. I could change it though if validation is preferred.